src/node_modules/app/tsservices/test.ts

```diff
@@ -0,0 +1,105 @@
+import * as ts from "typescript";
+import {src} from './testsrc';
+
+
+const LIBDTS = `// Standard types
+interface Date {}
+
+type Record<K extends keyof any, T> = {
+    [P in K]: T;
+};
+`;
+
+const srcname = "mysrc.ts";
+
+const host = createLanguageServiceHost(srcname, src);
+const service = ts.createLanguageService(host, ts.createDocumentRegistry());
+const semDiags = service.getSemanticDiagnostics(srcname);
+console.log(semDiags);
+const synDiags = service.getSyntacticDiagnostics(srcname);
+console.log(synDiags);
+
+
+const position = src.lastIndexOf('{') + 1;
+for (let p = position-20; p < src.length; p++) {
+	try {
+		console.log(src.slice(src.slice(0,p).lastIndexOf('\n'), p));
+		const info = service.getCompletionsAtPosition(srcname, p, {
+			//includeCompletionsWithInsertText: true,
+			//includeInsertTextCompletions: true,
+			//includeExternalModuleExports: true,
+			//includeAutomaticOptionalChainCompletions: true
+		});
+		//console.log(info && info.entries.map(i => i.name).join(', '));
+		console.log(info && info.entries.map(i => {
+
+			const details = service.getCompletionEntryDetails(srcname, p, i.name, undefined, undefined, undefined);
+			const displayText = details.displayParts.map(i => i.text).join('');
+			return `${details.name} -> ${displayText}`;
+		}));
+		//console.log(info && info.entries);
+	}
+	catch (e) {
+		console.log(e);
+	}
+}
+
+function createLanguageServiceHost(name, source): ts.LanguageServiceHost {
```

Comment:
This seems a duplication of [`servicesHost`](https://github.com/nestauk/dapsboard/pull/47/files#diff-46c94ee3d38479c684836ab06820abdaR12), maybe we can import it from there?